### DISCUSSION TITLE
LDM Notes for February 28th, 2024

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#7983] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2024/LDM-2024-02-28.md

### Agenda

- Extensions

**Comment by TahirAhmadov** [comment]
> For either of these cases, it seems that the best solution is simply to give both ToX methods to overload resolution and let it sort out which one is preferred, erroring if neither is preferred.

Agree with this. This conflict likely indicates an issue with the code which the developer needs to address, not sweep under the rug.

**Reply by TahirAhmadov** [reply]
> ```cs
> class TableIDoNotOwn : IEnumerable<Item> { }
> 
> static class IEnumerableExtensions
> {
>     public int Count<T>(this IEnumerable<T> t);
> }
> 
> implicit extension MyTableExtensions for TableIDoNotOwn
> {
>     public int Count { get { ... } }
> }
> 
> // What happens here?
> var v = table.Count; // Let's get a read from LDM
> ```

My first thought is, this should be an error - again, this likely indicates an issue which needs to be addressed.

**Comment by MgSam** [comment]
I'm looking forward to extensions hopefully finally landing in v13! Possibly the longest in-the-oven time of any C# feature that I can remember! Since at least 2009!

**Reply by 333fred** [reply]
Static abstracts for numerics were first proposed internally in 2006, so we've got a few years to go.

**Reply by sab39** [reply]
NRT was a long time coming as well, I'm fairly sure sure I remember a public blog series on what would be involved in adding non-nullable reference types to C# around the same time that nullable value types were being added, which would be 2005 or so.

**Reply by Richiban** [reply]
Didn't Gafter say that he worked on discriminated unions for C# back in 2010? They're not the longest-running yet, but they definitely will be should they get released!
